plugins/storage/volume/storpool/src/main/java/com/cloud/hypervisor/kvm/storage/StorpoolStorageAdaptor.java

```diff
@@ -0,0 +1,384 @@
+import org.apache.log4j.Logger;
+
+import com.cloud.agent.api.to.DiskTO;
+import com.cloud.storage.Storage;
+import com.cloud.storage.Storage.ImageFormat;
+import com.cloud.storage.Storage.ProvisioningType;
+import com.cloud.storage.Storage.StoragePoolType;
+import com.cloud.utils.exception.CloudRuntimeException;
+import com.cloud.utils.script.OutputInterpreter;
+import com.cloud.utils.script.Script;
+
+@StorageAdaptorInfo(storagePoolType=StoragePoolType.SharedMountPoint)
+public class StorpoolStorageAdaptor implements StorageAdaptor {
+    public static void SP_LOG(String fmt, Object... args) {
+        try (PrintWriter spLogFile = new PrintWriter(new BufferedWriter(new FileWriter("/var/log/cloudstack/agent/storpool-agent.log", true)))) {
+            final String line = String.format(fmt, args);
+            String timeStamp = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss,ms").format(Calendar.getInstance().getTime());
+            spLogFile.println(timeStamp +" "+line);
+            spLogFile.flush();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static final Logger log = Logger.getLogger(StorpoolStorageAdaptor.class);
+
+    private static final Map<String, KVMStoragePool> storageUuidToStoragePool = new HashMap<String, KVMStoragePool>();
+
+    @Override
+    public KVMStoragePool createStoragePool(String uuid, String host, int port, String path, String userInfo, StoragePoolType storagePoolType) {
+        SP_LOG("StorpooolStorageAdaptor.createStoragePool: uuid=%s, host=%s:%d, path=%s, userInfo=%s, type=%s", uuid, host, port, path, userInfo, storagePoolType);
+
+        StorpoolStoragePool storagePool = new StorpoolStoragePool(uuid, host, port, storagePoolType, this);
+        storageUuidToStoragePool.put(uuid, storagePool);
+        return storagePool;
+    }
+
+    @Override
+    public KVMStoragePool getStoragePool(String uuid) {
+        SP_LOG("StorpooolStorageAdaptor.getStoragePool: uuid=%s", uuid);
+        return storageUuidToStoragePool.get(uuid);
+    }
+
+    @Override
+    public KVMStoragePool getStoragePool(String uuid, boolean refreshInfo) {
+        SP_LOG("StorpooolStorageAdaptor.getStoragePool: uuid=%s, refresh=%s", uuid, refreshInfo);
+        return storageUuidToStoragePool.get(uuid);
+    }
+
+    @Override
+    public boolean deleteStoragePool(String uuid) {
+        SP_LOG("StorpooolStorageAdaptor.deleteStoragePool: uuid=%s", uuid);
+        return storageUuidToStoragePool.remove(uuid) != null;
+    }
+
+    @Override
+    public boolean deleteStoragePool(KVMStoragePool pool) {
+        SP_LOG("StorpooolStorageAdaptor.deleteStoragePool: uuid=%s", pool.getUuid());
+        return deleteStoragePool(pool.getUuid());
+    }
+
+    private static long getDeviceSize(final String devPath) {
+        SP_LOG("StorpooolStorageAdaptor.getDeviceSize: path=%s", devPath);
+
+        if (getVolumeNameFromPath(devPath, true) != null) {
```

Comment:
We could invert this condition and add the return. This way we would reduce the indentation of the block.